backend/volunteer_hub/views.py

```diff
@@ -1,21 +1,133 @@
 from rest_framework.views import APIView
+from django.contrib.gis.db.models.functions import Distance
 from rest_framework.response import Response
 from rest_framework import status
 from rest_framework.permissions import AllowAny
-from .models import Volunteer
-from .serializers import VolunteerSerializer
+from django.contrib.postgres.search import SearchVector
+from django.db import models
+from django.db.models import Q
+from django.contrib.gis.geos import Point
+from django.contrib.gis.measure import D
+from .models import Volunteer, VolunteerTeam
+from .serializers import VolunteerSerializer, VolunteerTeamSerializer
 from drf_yasg.utils import swagger_auto_schema
 from drf_yasg import openapi
 
 
 class VolunteerListView(APIView):
-    permission_classes = [AllowAny]  # Allow any user to access this view
+    permission_classes = [AllowAny]
 
     @swagger_auto_schema(
-        operation_description="Retrieve a list of active volunteers",
+        operation_description="Retrieve a list of active volunteers with optional filtering and searching.",
+        manual_parameters=[
+            openapi.Parameter(
+                'thana', openapi.IN_QUERY, description="Filter by Thana ID", type=openapi.TYPE_INTEGER
+            ),
+            openapi.Parameter(
+                'district', openapi.IN_QUERY, description="Filter by District ID", type=openapi.TYPE_INTEGER
+            ),
+            openapi.Parameter(
+                'address', openapi.IN_QUERY, description="Full-text search on address", type=openapi.TYPE_STRING
+            ),
+            openapi.Parameter(
+                'latitude', openapi.IN_QUERY, description="Latitude for geospatial search", type=openapi.TYPE_NUMBER
+            ),
+            openapi.Parameter(
+                'longitude', openapi.IN_QUERY, description="Longitude for geospatial search", type=openapi.TYPE_NUMBER
+            ),
+        ],
         responses={200: VolunteerSerializer(many=True)},
     )
     def get(self, request, *args, **kwargs):
         volunteers = Volunteer.objects.filter(is_active=True)
+
+        # Filtering by Thana and District
+        thana_id = request.query_params.get('thana')
+        district_id = request.query_params.get('district')
+
+        if thana_id:
+            volunteers = volunteers.filter(thana_id=thana_id)
+        if district_id:
+            volunteers = volunteers.filter(district_id=district_id)
+
+        address_query = request.query_params.get('address')
+        if address_query:
+            volunteers = volunteers.filter(
+                address__icontains=address_query
+            ).distinct()
+
+        latitude = request.query_params.get('latitude')
+        longitude = request.query_params.get('longitude')
+        if latitude and longitude:
+            user_location = Point(float(longitude), float(latitude), srid=4326)
```

Comment:
Is this project use the WGS 84 coordinating system? If yes, then it's fine to use `srid=4326`.